### PR TITLE
Update PR and Release Docs

### PR DESCRIPTION
**Pull request by medley56**
Add .venv to .gitignore
Add a CODEOWNERS file to automate adding PR reviewers
Update PR guidelines and release documentation

### CODE REVIEW

File: .github/CODEOWNERS
Comment (greglucas) [inline .github/CODEOWNERS:1]: Nice, I didn't know about this!

File: docs/source/developers.md
Comment (greglucas) [inline docs/source/developers.md:25]: Maybe add this into a PR template with checkboxes for contributors?
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/.github/pull_request_template.md

I do like how nice and contained this is.

**Review by greglucas — APPROVED** [review]

File: docs/source/developers.md
Comment (medley56) [reply docs/source/developers.md]: Good suggestion on the PR template. I didn't know about that.

I need another Approval since this changed the PR after the last approval. Then I think this is ready.

**Review by greglucas — APPROVED** [review]

**Pull request merged**
